src/scala/ly/stealth/mesos/kafka/Rebalancer.scala

```diff
@@ -121,6 +142,39 @@ class Rebalancer {
     reassignment.toMap
   }
 
+  private def getRealignments(brokerIds: Seq[Int], topics: util.List[String], assignment: Map[TopicAndPartition, Seq[Int]], replicas: Int = -1, fixedStartIndex: Int = -1): Map[TopicAndPartition, Seq[Int]] = {
+    var realignment : Map[TopicAndPartition, Seq[Int]] = new mutable.HashMap[TopicAndPartition, List[Int]]()
+
+    val rand = new Random
+    var replacementReplicaIndex = if (fixedStartIndex != -1) fixedStartIndex else rand.nextInt(brokerIds.size)
+    assignment.foreach { entry =>
+      val topic = entry._1
+      val members = entry._2
+      val rf: Int = if (replicas != -1) replicas else members.size
+
+      if((rf == members.size) && ((brokerIds intersect members).size == members.size)) {
+        realignment += (topic -> members)
```

Comment:
It occurred to me last night that I could avoid putting these in the realignment map entirely and then the controller would never consider them.  I would need to scrub them from the assignment map on line 57 above too, but this seems more efficient.  For example, it would let us shrink the cluster by a few brokers and issue a realignment against all partitions without the controller exploding.